cost-analyzer/charts/prometheus/values.yaml

```diff
@@ -1216,6 +1216,62 @@ serverFiles:
             regex: (.+)
             action: replace
 
+      # A scrape configuration for running Prometheus on a Kubernetes cluster.
+      # This uses separate scrape configs for cluster components (i.e. API server, node)
+      # and services to allow each to use different authentication configs.
+      #
+      # Kubernetes labels will be added as Prometheus labels on metrics via the
+      # `labelmap` relabeling action.
+
+      - job_name: 'kubernetes-nodes'
+
+        # Default to scraping over https. If required, just disable this or change to
+        # `http`.
+        scheme: https
+
+        # This TLS & bearer token file config is used to connect to the actual scrape
+        # endpoints for cluster components. This is separate to discovery auth
+        # configuration because discovery & scraping are two separate concerns in
+        # Prometheus. The discovery auth config is automatic if Prometheus runs inside
+        # the cluster. Otherwise, more config options have to be provided within the
+        # <kubernetes_sd_config>.
+        tls_config:
+          ca_file: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+          # If your node certificates are self-signed or use a different CA to the
+          # master CA, then disable certificate verification below. Note that
+          # certificate verification is an integral part of a secure infrastructure
+          # so this should only be disabled in a controlled environment. You can
+          # disable certificate verification by uncommenting the line below.
+          #
+          insecure_skip_verify: true
+        bearer_token_file: /var/run/secrets/kubernetes.io/serviceaccount/token
+
+        kubernetes_sd_configs:
+          - role: node
+
+        relabel_configs:
+          - action: labelmap
+            regex: __meta_kubernetes_node_label_(.+)
+          - target_label: __address__
+            replacement: kubernetes.default.svc:443
+          - source_labels: [__meta_kubernetes_node_name]
+            regex: (.+)
+            target_label: __metrics_path__
+            replacement: /api/v1/nodes/$1/proxy/metrics
+
+        metric_relabel_configs:
+          - source_labels: [ __name__ ]
+            regex: (kubelet_volume_stats_used_bytes)
```

Comment:
IIRC `kubelet_volume_stats_used_bytes` is in alpha status as a metric. Based on the [documentation](https://kubernetes.io/docs/concepts/cluster-administration/system-metrics/#metric-lifecycle), that means that the metric is subject to changes or removal at any time.

Is the code you've written using this metric resilient to the removal/change of the metric? In other words, if the metric were to disappear tomorrow, would the rest of the Asset/Disk pipeline still continue to function as it does today?

For product/eng leadership (@dwbrown2, @nikovacevic): Do we have guidelines on building Kubecost functionality on top of alpha/beta K8s functionality? For context, we've previously had contention over a beta->v1 migration/breaking change with cluster-turndown's CRD: https://github.com/kubecost/cost-analyzer-helm-chart/pull/1444#issuecomment-1145092419.

Reply:
Are we sure this metric is still alpha? I haven't chased all the way but found this: https://github.com/kubernetes/kubernetes/pull/51553

I think these decisions should be situational. I personally think adding a metric here is valuable as with our past experience it was accurate the vast majority of the time. I do think we should be resilient in the application experience to it being not available on a cluster. We should also consider marking this feature beta in the UI.

Reply:
I have a K8s v1.22 GKE cluster:
```
→ k version 
...
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.12-gke.2300", GitCommit:"e55564cf3a1384026a54920174977659c8c56a50", GitTreeState:"clean", BuildDate:"2022-08-16T09:24:51Z", GoVersion:"go1.16.15b7", Compiler:"gc", Platform:"linux/amd64"}
...
```

When I look for this metric (its a standard kubelet metric, so at a node's kubelet's /metrics, I see that its an ALPHA status:

```
kubectl proxy --port 8033

curl 'localhost:8033/api/v1/nodes/gke-michael-dev-pool-1-92180698-v5zo/proxy/metrics' | grep -i 'kubelet_volume_stats_used_bytes'

...
# HELP kubelet_volume_stats_used_bytes [ALPHA] Number of used bytes in the volume
# TYPE kubelet_volume_stats_used_bytes gauge
```

Do you know if it has moved out of ALPHA status since 1.22? That PR you linked was merged before v1.22.

Reply:
I think it's ok to put an alpha metric in but concur that resiliency when the metric is not available is important.